### PR TITLE
Add UninstallTool

### PR DESCRIPTION
**Pull request by nlathia**

### CODE REVIEW

**Comment by nlathia** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/gradientlabs-ai/go-client/34?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#34** <a href="https://app.graphite.dev/github/pr/gradientlabs-ai/go-client/34?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/gradientlabs-ai/go-client/34?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#33** <a href="https://app.graphite.dev/github/pr/gradientlabs-ai/go-client/33?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#32** <a href="https://app.graphite.dev/github/pr/gradientlabs-ai/go-client/32?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#31** <a href="https://app.graphite.dev/github/pr/gradientlabs-ai/go-client/31?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#30** <a href="https://app.graphite.dev/github/pr/gradientlabs-ai/go-client/30?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: tool_uninstall.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+func (c *Client) UninstallTool(ctx context.Context, toolID string) error {
+	rsp, err := c.makeRequest(ctx, http.MethodDelete, fmt.Sprintf("tools/%s", toolID), nil)
+	if err != nil {
+		return err
+	}
+	defer rsp.Body.Close()
+
+	if err := responseError(rsp); err != nil {
+		return err
+	}
```
Comment (twindebank) [inline tool_uninstall.go:22]: nit: could tidy the code a bit if this block were moved inside of `makeRequest`, seems to be used in the same way every time

Comment (nlathia) [reply tool_uninstall.go:22]: agreed, we can do this separately

**Review by twindebank — APPROVED** [review]

**Comment by nlathia** [comment]
### Merge activity

* **Dec 16, 1:27 PM EST**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/gradientlabs-ai/go-client/34).

**Pull request merged**